### PR TITLE
Prevent double payment request init on checkout page

### LINKED ISSUE #8266: Payment Request is initialized twice on checkout page
### Describe the bug
The same issue has been reported with Woo Stripe [here](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2471).

The payment request button is initialized twice due to the presence of both a "global" init and an init that fires on the `updated_checkout` event. The `updated_checkout` event is triggered when the checkout page loads, so the initialization is executed twice, leading to 2 ajax calls to `get_cart_details()` every time the checkout page loads.

**First init:**
https://github.com/Automattic/woocommerce-payments/blob/d20b82ff9d1e5d7128c05bd70d65232e41083032/client/payment-request/index.js#L655-L657

**Second init:**
https://github.com/Automattic/woocommerce-payments/blob/d20b82ff9d1e5d7128c05bd70d65232e41083032/client/payment-request/index.js#L662-L667

### To Reproduce

1. Enable Payment Request buttons
2. Add any product to cart
3. Open up developer console and go to "network" tab
4. Go to checkout
5. Filter for `get_cart_details`
6. Notice 2 `wc-ajax=wcpay_get_cart_details` calls

### Actual behavior

Screenshot 2024-02-23 at 10 39 25 AM

### Expected behavior

`wcpayPaymentRequest.init()` should only be triggered once when the checkout page loads and when the checkout is updated.

### Additional context

We should be able forgo the first `wcpayPaymentRequest.init()` on the checkout page and rely on the `updated_checkout` event for initialization on that page. We'll just need to check that there's not a scenario when that event doesn't fire when the checkout page loads, which I think is unlikely.

7783809-zen
p1701259742368559-slack-C7U3Y3VMY

### PR DESCRIPTION
**Pull request by gpressutto5**
Fixes #8266

#### Changes proposed in this Pull Request

The checkout page always triggers the `updated_checkout` event, which initializes the payment request here https://github.com/Automattic/woocommerce-payments/pull/8282/files#diff-3b09c95dc3e8c6f0dd75f523bc8fccba95d8308a909d8641c5248c92134b2cacL664-L666. Because of this we were always initializing the payment request twice in the checkout page. With these changes we are removing the first initialization and letting `updated_checkout` trigger it.

#### Testing instructions

Smoke test Payment Request button in checkout, product page, and cart page.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8282` or branch name `fix-double-init-payment-request` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix-double-init-payment-request) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8a4e587c19eff59c5b903ffb0044f446e2195a50
- Build time: 2024-02-29 18:42:48 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-request/index.js [hunk 1]
```diff
@@ -653,7 +653,10 @@ jQuery( ( $ ) => {
 		},
 	};
 
-	wcpayPaymentRequest.init();
+	// We don't need to initialize payment request on the checkout page now because it will be initialized by updated_checkout event.
+	if ( ! wcpayPaymentRequestParams.is_checkout_page ) {
+		wcpayPaymentRequest.init();
+	}
```
Comment (mdmoore) [inline client/payment-request/index.js:663]:
```suggestion
if ( wcpayPaymentRequestParams.is_pay_for_order ) {
	wcpayPaymentRequest.init();
}
```

**Review by mdmoore — CHANGES_REQUESTED** [review]
This does the trick for preventing the extra request on the checkout page. I noticed that the PRB doesn't appear on the Pay for Order page though. That's because `is_checkout()` is `true` on the Pay for Order page, so I think we'll need to init the PRB specifically for that page.

File: client/payment-request/index.js (hunk 1 above)
Comment (duotive) [reply client/payment-request/index.js:663]: @mdmoore The payment request buttons are also present on product pages. Should we also have a check for product pages using existing variables in the localized script, considering that checkout triggers will not fire there?

`
if ( wcpayPaymentRequestParams.is_pay_for_order || wcpayPaymentRequestParams.is_product_page ) {
`

Comment (mdmoore) [reply client/payment-request/index.js:663]: @duotive the `! wcpayPaymentRequestParams.is_checkout_page` condition ensures that the payment request button is initialized on product pages.

Comment (gpressutto5) [reply client/payment-request/index.js:663]: Nice catch! I'm updating that with `(! wcpayPaymentRequestParams.is_checkout_page || wcpayPaymentRequestParams.is_pay_for_order)`

**Review by mdmoore — APPROVED** [review]
LGTM! 🚢

**Pull request merged**